core/src/main/java/hudson/ProxyConfiguration.java

```diff
@@ -259,6 +259,37 @@ public PasswordAuthentication getPasswordAuthentication() {
         return con;
     }
 
+    /**
+     * This method should be used wherever {@link URL#openConnection()} to internet URLs is invoked directly and a redirection is possible.
+     */
+    public static URLConnection openAllowsRedirection(URL url) throws IOException {
```

Comment:
If it is public, add `@since TODO`. If you do not want it to be fully public, add `@Restricted(NoExternalUse.class)`

Reply:
This method should be equals to open, so i will add `@since TODO`